### PR TITLE
make sidecar resource limit configurable

### PR DESCRIPTION
**Pull request by ajayOO8**
Currently the default limits/request of the hacheck, uwsgi containers is huge for a sidecar containers.
making these configuration configurable.
for now I have set the default limit as same as the current limit.

### CODE REVIEW

File: paasta_tools/kubernetes_tools.py [hunk 1]
```diff
@@ -1250,15 +1256,23 @@ def get_resource_requirements(self) -> V1ResourceRequirements:
...
+        sidecar_name: str,
+        system_paasta_config: SystemPaastaConfig,
     ) -> V1ResourceRequirements:
         config = self.config_dict.get("sidecar_resource_requirements", {}).get(
             sidecar_name, {}
         )
+        sidecar_requirements_config = (
+            system_paasta_config.get_sidecar_requirements_config().get(
+                sidecar_name,
+                {"cpu": 0.1, "memory": "1024Mi", "ephemeral-storage": "256Mi"},
```
[the remark this answers is not in the document]
Comment (ajayOO8) [reply paasta_tools/kubernetes_tools.py]: added the CONSTANT.
i wanted to have a default override layer for limits, thats why i put it here

File: paasta_tools/utils.py [hunk 2]
```diff
@@ -2091,6 +2092,9 @@ def __repr__(self) -> str:
     def get_spark_use_eks_default(self) -> bool:
         return self.config_dict.get("spark_use_eks_default", False)
 
+    def get_sidecar_requirements_config(self) -> Mapping[str, Mapping[str, str]]:
```
[the remark this answers is not in the document]
Comment (ajayOO8) [reply paasta_tools/utils.py]: updated.

File: paasta_tools/kubernetes_tools.py [hunk 3]
```diff
@@ -201,6 +202,7 @@
 DEFAULT_USE_PROMETHEUS_CPU = False
 DEFAULT_USE_PROMETHEUS_UWSGI = True
 DEFAULT_USE_RESOURCE_METRICS_CPU = True
+DEFAULT_SIDECAR_REQUEST = {"cpu": 0.1, "memory": "1024Mi", "ephemeral-storage": "256Mi"}
```
Comment (nemacysts) [inline paasta_tools/kubernetes_tools.py:205]:
```suggestion
DEFAULT_SIDECAR_REQUEST: KubeContainerResourceRequest = {"cpu": 0.1, "memory": "1024Mi", "ephemeral-storage": "256Mi"}
```

**Review by nemacysts — COMMENTED** [review]
generally lgtm, but you'll want to also update the callsite for where the gunicorn exporter gets its requirements :)

(it should behave pretty similarly to the uwsgi exporter)

**Review by nemacysts — APPROVED** [review]

**Pull request merged**